### PR TITLE
typescript-inversify: improve check for required parameters, support multiple media types

### LINKED ISSUE #3838: [REQ] [typescript-inversify] Three small changes and one big?
We have adapted the typescript-inversify template for our own use. Some changes are definitely worth integrating into upstream. As per the contributing guidelines, let us discuss these changes. Pull request should follow.

I think for fixing these small issues, one pull request would be ok. I would also include a fix of #3618 

## Falsy Required Parameters
When a required parameter is falsy, an error is thrown. But `""` and `false` can be valid values. Cf. https://github.com/OpenAPITools/openapi-generator/blob/9cc7bd15f2884b12b373b172fa175063897724ff/modules/openapi-generator/src/main/resources/typescript-inversify/api.service.mustache#L63-L65

## Multiple Media Types
The Accept header can only contain one media type in the current implementation. Cf. https://github.com/OpenAPITools/openapi-generator/blob/9cc7bd15f2884b12b373b172fa175063897724ff/modules/openapi-generator/src/main/resources/typescript-inversify/api.service.mustache#L138-L151

## WithInterfaces
The template provides the `withInterfaces` option. When you enable it, `apis.mustache` still generates exports for the implementations, not the interfaces. *Should this be changed?* (Would be a breaking change I guess.)

## Class Based Service Identificators
*This is a major breaking change, so I'm unsure whether it should be integrated.*

Currently the dependency injection provided by the template uses string based service identifiers. This means that type checking for the services has to be setup manually at the point of composition. Especially when interfaces are changing, this can lead to overlooked problems. See [inversify issue 911](https://github.com/inversify/InversifyJS/issues/911) for a discussion.

The solution we came up with, is using abstract classes as service identifiers. This way the composition of the services changes from
```typescript
container.bind<IHttpClient>("IApiHttpClient").to(HttpClient).inSingletonScope();
container.bind<IApiConfiguration>("IApiConfiguration").to(ApiConfiguration).inSingletonScope();
```
to
```typescript
container.bind(IHttpClient).to(HttpClient).inSingletonScope();
container.bind(IApiConfiguration).to(ApiConfiguration).inSingletonScope();
```
and the types will be checked automatically.

After typescript compilation the abstract classes are still there, but because they don’t contain any implementation, they don’t use more resources than the strings.

I could create a separate pull-request for this change, but only if there is a genuine chance that it will be included.

### PR DESCRIPTION
**Pull request by bodograumann**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, ~`4.1.x`~, ~`5.0.x`~. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Solves two small issues as described in #3838 (Falsy Required Types and Multiple Media Types) as well as the issue in #3618 (IHttpClient interface).

There is no testing setup for typescript-inversify. These changes are very simple though.

fixes #3838, fixes #3849 
 
### Technical Committee ###
@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10) @akehir (2019/07)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/typescript-inversify/api.service.mustache [hunk 1]
```diff
@@ -60,7 +60,7 @@ export class {{classname}} {
     public {{nickname}}({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}observe: any = 'body', headers: Headers = {}): {{#usePromise}}Promise{{/usePromise}}{{^usePromise}}Observable{{/usePromise}}<any> {
 {{#allParams}}
 {{#required}}
-        if (!{{paramName}}){
+        if ({{paramName}} == null){
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-inversify/api.service.mustache]:
```suggestion
        if ({{paramName}} === null || {{paramName}} === undefined){
```

**Comment by macjohnny** [comment]
@bodograumann thanks for your PR

File: modules/openapi-generator/src/main/resources/typescript-inversify/api.service.mustache (hunk 1 above)
Comment (bodograumann) [reply modules/openapi-generator/src/main/resources/typescript-inversify/api.service.mustache]: Is there a specific reason for this change?
The semantics for both versions are identical. Cf. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#Loose_equality_using

Comment (macjohnny) [reply modules/openapi-generator/src/main/resources/typescript-inversify/api.service.mustache]: strict equality check is more explicit and easier to read, as it indicates that both cases are considered. I agree this does not change the functionality, but only the readability

Comment (bodograumann) [reply modules/openapi-generator/src/main/resources/typescript-inversify/api.service.mustache]: Ok, sure, that makes sense.

**Comment by bodograumann** [comment]
You are absolutely right.

**Pull request merged**